ast/src/expression/value.rs

```diff
@@ -19,32 +19,43 @@ use tendril::StrTendril;
 use super::*;
 use crate::{Char, CharValue};
 
+/// A literal expression.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum ValueExpression {
     // todo: deserialize values here
+    /// An address literal, e.g., `aleo1qnr4dkkvkgfqph0vzc3y6z2eu975wnpz2925ntjccd5cfqxtyu8sta57j8`.
     Address(
         #[serde(with = "leo_span::tendril_json")] StrTendril,
         #[serde(with = "leo_span::span_json")] Span,
     ),
+    /// A boolean literal, either `true` or `false`.
     Boolean(
         #[serde(with = "leo_span::tendril_json")] StrTendril,
         #[serde(with = "leo_span::span_json")] Span,
     ),
+    /// A char literal, e.g., `'a'`, representing a single unicode code point.
     Char(CharValue),
+    /// A field literal, e.g., `42field`.
+    /// That is, an unsigned number followed by the keyword `field`.
     Field(
         #[serde(with = "leo_span::tendril_json")] StrTendril,
         #[serde(with = "leo_span::span_json")] Span,
     ),
+    /// A group literal, either single or tuple.
```

Comment:
I would replace
single => product
tuple => affine
to follow more typical elliptic curve terminology and also ABNF grammar terminology.